monai/transforms/croppad/array.py

```diff
@@ -1344,8 +1345,19 @@ def randomize(
             _shape = image.peek_pending_shape() if isinstance(image, MetaTensor) else image.shape[1:]
         if _shape is None:
             raise ValueError("label or image must be provided to infer the output spatial shape.")
+        img_shape = None
```

Comment:
Thanks for the PR!
But I think we could have a check for `img_shape` and `label_shape` if they are not equal, then we may raise a warning in the call.

Reply:
There should be a warning about this yes.